truss/templates/control/control/application.py

```diff
@@ -1,71 +1,113 @@
+import asyncio
 import logging
 import re
 from pathlib import Path
 from typing import Dict
 
+import httpx
 from endpoints import control_app
-from flask import Flask
-from helpers.errors import PatchApplicatonError
+from fastapi import FastAPI
+from fastapi.responses import JSONResponse
+from helpers.errors import ModelLoadFailed, PatchApplicatonError
 from helpers.inference_server_controller import InferenceServerController
 from helpers.inference_server_process_controller import InferenceServerProcessController
+from helpers.inference_server_starter import async_inference_server_startup_flow
 from helpers.logging import setup_logging
 from helpers.truss_patch.model_container_patch_applier import ModelContainerPatchApplier
-from werkzeug.exceptions import HTTPException
+from starlette.datastructures import State
 
 
-def create_app(base_config: Dict):
-    app = Flask(__name__)
+async def handle_patch_error(_, exc):
+    error_type = _camel_to_snake_case(type(exc).__name__)
+    return JSONResponse(
+        content={
+            "error": {
+                "type": error_type,
+                "msg": str(exc),
+            }
+        }
+    )
+
+
+async def generic_error_handler(_, exc):
+    return JSONResponse(
+        content={
+            "error": {
+                "type": "unknown",
+                "msg": f"{type(exc)}: {exc}",
+            }
+        }
+    )
+
+
+async def handle_model_load_failed(_, error):
+    # Model load failures should result in 503 status
+    return JSONResponse({"error": str(error)}, 503)
 
+
+def create_app(base_config: Dict):
+    app_state = State()
     setup_logging()
+    app_logger = logging.getLogger(__name__)
+    app_state.logger = app_logger
+
+    for k, v in base_config.items():
+        setattr(app_state, k, v)
 
-    app.logger.setLevel(logging.DEBUG)
-    app.config.update(base_config)
-    app.config[
-        "inference_server_process_controller"
-    ] = InferenceServerProcessController(
-        app.config["inference_server_home"],
-        app.config["inference_server_process_args"],
-        app.config["inference_server_port"],
-        app_logger=app.logger,
+    app_state.inference_server_process_controller = InferenceServerProcessController(
+        app_state.inference_server_home,
+        app_state.inference_server_process_args,
+        app_state.inference_server_port,
+        app_logger=app_logger,
     )
 
+    limits = httpx.Limits(max_keepalive_connections=8, max_connections=32)
+    app_state.proxy_client = httpx.AsyncClient(
```

Comment:
Is there a way to set a long default timeout here? We noticed `httpx.ReadTimeout` errors when this was running, so I wonder if it were for the lack of good configurations on the client.